_posts/2020-08-26-#19339.md

```diff
@@ -0,0 +1,117 @@
+  - This also contrasts with user- or client-specific rules. These may
+  vary by client or even by call (e.g. arguments passed into RPCs).
+
+* Pre-19339, ATMP accepts an argument, `const CAmount& nAbsurdFee`, and
+  enforces it as a maximum fee amount; this protects users from paying
+  “absurdly high” fees resulting from unintentional errors.  ATMP returns a
+  `TxValidationResult::TX_NOT_STANDARD` error for absurd fees, which seems to
+  suggest that rejecting absurd fees is part of policy.
+
+* However, `nAbsurdFee` is only used for transactions originating from the
+  node’s clients (RPC and wallet).  When a node receives a transaction from its
+  peers, it uses an empty value for `nAbsurdFee` when calling ATMP (you can see
+  the values of `nAbsurdFee` removed in the [Remove absurdfee from
+  accepttomempool](https://github.com/bitcoin-core-review-club/bitcoin/commit/67bb7148)
+  commit).  Furthermore, there is no way for a node operator to set the maximum
+  fee, e.g. through a config setting or command line argument.  The wallet, on
+  the other hand, allows the user to set a maximum fee using `-maxtxfee`.
+
+* [#19339](https://github.com/bitcoin/bitcoin/pull/19339) redelegates fee
+  checking from the mempool to the clients in order to make mempool logic less
+  user-specific.  The main observable behavior change is in the error message,
+  which is updated to include RPC.
+
+  * This only changes RPCs
+    [`testmempoolaccept`](https://github.com/bitcoin/bitcoin/blob/3ab2582c/src/rpc/rawtransaction.cpp#L858)
+    and
+    [`sendrawtransaction`](https://github.com/bitcoin/bitcoin/blob/3ab2582c/src/rpc/rawtransaction.cpp#L800).
+    There are no changes to wallet (reviewers may want to verify that this is not
+    an issue).
+
+  * Note that clients often use
+    [`BroadcastTransaction()`](https://github.com/bitcoin/bitcoin/blob/3ab2582c/src/node/transaction.cpp#L16)
+    to interface with ATMP; it also takes `max_tx_fee` to pass in as `nAbsurdFee`.
+    This PR adds fee checking to `BroadcastTransaction()` by adding a dry run of
+    ATMP before the real submission.
+
+* To be careful to preserve the fee protection offered to the user, the PR
+  includes a few intermediate stages.  The author recommends reviewing each
+  commit individually and confirming that the behavior is identical at each
+  stage.
+
+  1. [Catch mempool rejections based on `AbsurdFee` and return
+     `max-fee-exceeded` instead of `absurdly-high-fee` for high
+     fees](https://github.com/bitcoin-core-review-club/bitcoin/commit/18f7e1fb).
+
+  2. [In `BroadcastTransaction`, first dry-run ATMP and check the fee that is
+     returned](https://github.com/bitcoin-core-review-club/bitcoin/commit/a872d8d3).
+     If the fee is below `max_tx_fee`, send the transaction for real.
+
+  3. [Ignore `nAbsurdFee` in
+     ATMP](https://github.com/bitcoin-core-review-club/bitcoin/commit/24094623).
+
+  4. [Remove `nAbsurdFee` from
```

Comment:
The commit name links in this ordered list would be more readable without the `code` markup, which combined with the link markup and the regular text doesn't render super well imo.